packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -771,11 +775,29 @@ abstract class State<T extends StatefulWidget> {
   /// framework might not schedule a [build] and the user interface for this
   /// subtree might not be updated to reflect the new state.
   ///
+  /// Generally it is recommended that the `setState` method only be used to
+  /// wrap the actual changes to the state, not any computation that might be
+  /// associated with the change. For example, here a value used by the [build]
+  /// function is incremented, and then the change is written to disk, but only
+  /// the increment is wrapped in the `setState`:
+  ///
+  /// ```dart
+  /// Future<Null> _incrementCounter() async {
+  ///   setState(() {
+  ///     _counter++;
+  ///   });
+  ///   final String dir = await PathProvider.getApplicationDocumentsDirectory();
+  ///   await new File('$dir/counter.txt').writeAsString('$_counter');
```

Comment:
Is it better to await the writeAsString as well, so that I/O errors can be caught?